docs/proposals/13-automated-seed-management.md

```diff
@@ -0,0 +1,289 @@
+
+Example Seed status with `capacity` and `allocatable` fields:
+
+```yaml
+status:
+  capacity:
+    shoots: "100"
+    persistent-volumes: "200" # Built-in resource
+    azure.provider.extensions.gardener.cloud/load-balancers: "30" # Custom resource advertised by an Azure-specific plugin
+  allocatable:
+    shoots: "100"
+    persistent-volumes: "197" # 3 persistent volumes are reserved for Gardener
+    azure.provider.extensions.gardener.cloud/load-balancers: "300"
+```
+
+Example Shoot `resourceRequirements` field requesting a built-in and a custom resource:
+
+```yaml
+spec:
+  resourceRequirements:
+    requests:
+      persistent-volumes: 20
+      azure.provider.extensions.gardener.cloud/load-balancers: 10
+```
+
+### Gardenlet Configuration
+
+As mentioned above, the total resource capacity for built-in resources such as the number of shoots is specified as part of the `gardenlet` configuration, not in the Seed spec. The `gardenlet` configuration itself could be specified in the spec of the newly introduced [ShootedSeed](#shootedseeds) resource, which also contains the Seed template. Here it is assumed that in the future this could become the recommended and most widely used way to manage seeds. If the same `gardenlet` is responsible for multiple seeds, they would all share the same capacity settings.
+
+To specify the total resource capacity for built-in resources, as well as the amount of such resources reserved for Gardener, the 2 new fields `resources.capacity` and `resources.reserved` are introduced in the `GardenletConfiguration` resource. The `gardenlet` seed controller would then initialize the `capacity` and `allocatable` fields in the seed status as follows:
+
+* The `capacity` value is set to the configured `resources.capacity`.
+* The `allocatable` value is set to the configured `resources.capacity` minus `resources.reserved`. 
+
+Example `GardenletConfiguration` with `resources.capacity` and `resources.reserved` field:
+
+```yaml
+resources:
+  capacity:
+    shoots: 100
+    persistent-volumes: 200
+  reserved:
+    persistent-volumes: 3
+```
+
+### Custom Resources and Resource Plugins
+
+To handle custom resources, including provider-specific resources, a custom resource plugin framework similar to Kubernetes [Device Plugins](https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/) could be introduced to advertise custom resources to the `gardenlet`.
+
+Custom resource plugins could be deployed as sidecars in the `gardenlet` pod, so that they would be able to register and communicate with `gardenlet` via gRPC. Such resources might include for example load balancers on Azure, and other similar resources that may require provider-specific initialization or communication with the cloud provider.
```

Comment:
I could also remove the entire section or just mention that this could be the subject of a future GEP. A provider-specific resource such as `azure/load-balancers` might not need any plugin, unless a special initialization or communication with the cloud provider is required - and in fact I am not really sure if this is the case for Azure load balancers. So perhaps we don't need any more elaborate extensibility / plugin concept at least for the time being?

Reply:
Yes, I would probably exclude it for now. Load balancers are anyways something that are not provider-specific, i.e., the gardenlet can understand how many of them in the cluster exist and whether there is still capacity for more.

Reply:
Done as discussed above.